pkg/kubelet/kubelet_network.go

```diff
@@ -124,6 +124,9 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" || kl.flannelExperimentalOverlay {
 		return false
 	}
+	if kl.networkPlugin != nil && !kl.networkPlugin.Capabilities().Has(network.NET_PLUGIN_CAPABILITY_CLOUD_BASED) {
```

Comment:
Propose naming this `NET_PLUGIN_CAPABILITY_NODE_READINESS`, and we should glog.Warningf if the network provider is not validating that nodes are ready before allowing them to be scheduled.  It seems this is particularly important with NetworkPolicy!